az-digital/az_quickstart
# Fixes #2619 News export API empty value handling

## Description
For some of the API fields, `az_quickstart/export/az_news/v1/az_news.json` represents empty values as  `[ ]`, when the value of the equivalent non-empty field is normally a json object containing the field properties (this includes fields like `field_az_media_thumbnail_image`, `field_az_enterprise_attributes` and `field_az_media_image`.)

This PR changes handling of those fields to produce `null` rather than `[ ]` when those fields are empty. This is to increase data type consistency of the API.

Discussion for review: is `null` or an empty object (rather than an array) preferable?

**update: the PR has been updated to instead provide an empty object `{  }` , rather than `null`.**

## Related issues
Closes #2619 

## How to test

- enable `az_news_export`  `az_enterprise_attributes_import`
- Create news data
- Verify some data has images and others do not
- Verify that some data has enterprise attributes and others do not
- Visit `az_quickstart/export/az_news/v1/az_news.json` and verify that empty field values produce `null` rather than `[ ]` 

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
